spi/identity-hub-spi/src/main/java/org/eclipse/dataspaceconnector/identityhub/credentials/VerifiableCredentialsJwtUtils.java

```diff
@@ -0,0 +1,69 @@
+/*
+ *  Copyright (c) 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.identityhub.credentials;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.nimbusds.jwt.SignedJWT;
+import org.eclipse.dataspaceconnector.identityhub.credentials.model.VerifiableCredential;
+import org.eclipse.dataspaceconnector.spi.result.Result;
+
+import java.util.AbstractMap;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Util class to manipulate VerifiableCredentials with JWTs.
+ */
+public class VerifiableCredentialsJwtUtils {
+    private static final String VERIFIABLE_CREDENTIALS_KEY = "vc";
+    private ObjectMapper objectMapper;
+
+    public VerifiableCredentialsJwtUtils(ObjectMapper objectMapper) {
+        this.objectMapper = objectMapper;
+    }
+
+    /**
+     * Extract credentials from a JWT. The credential is represented with the following format
+     * "credentialId" : {
+     *     "vc": {
+     *         "credentialSubject": {
+     *             // some claims about the subject.
+     *         }
+     *     }
+     *     "iss": "issuer-value",
+     *     "sub": "subject-value",
+     *     // other JWT claims
+     * }
+     * The representation is used to support any type of <a href="https://www.w3.org/TR/vc-data-model">verifiable credentials</a> .
+     * When applying policies, the policy engine might need to access the issuer of the claim. That's why the JWT claims are included.
+     *
+     * @param jwt SignedJWT containing a verifiableCredential in its payload.
+     * @return VerifiableCredential represented as Map.Entry<String, Object>.
+     */
+    public Result<Map.Entry<String, Object>> extractCredential(SignedJWT jwt) {
```

Comment:
nit: leveraging a `Map.Entry` to represent a 2-tuple is somewhat confusing and semantically dubious, since this does not represent  a Map Entry. Creating a dedicated class should be easy enough, and make code more legible.

Reply:
As discussed, using `Map.Entry<String, Object>` matches the API of the `CredentialsVerifier` (`Result<Map<String, Object>> getVerifiedCredentials(DidDocument didDocument)`). Leaving it like it is.